packages/core/datadog-sourcemaps.gradle

```diff
@@ -0,0 +1,139 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+def reactConfig = project.hasProperty("react") ? project.react : [:]
+def reactRoot = file(reactConfig.root ?: "../../")
+
+def datadogConfig = new Properties()
+def datadogConfigFile = new File(reactRoot, "datadog-sourcemaps.properties")
+
+if (datadogConfigFile.exists()) {
+    datadogConfigFile.withInputStream {
+        datadogConfig.load(it)
+    }
+}
+
+afterEvaluate {
+
+    def androidExtension = extensions.findByName("android")
+
+    if (androidExtension == null) {
+        throw new GradleException("Android extension is missing.")
+    }
+
+    // TODO service name per variant?
+    def serviceName = datadogConfig.SERVICE_NAME ?: androidExtension.defaultConfig.applicationId
```

Comment:
Question: should we be able to provide this as an environment variable to get a similar behaviour on iOS and Android?

Reply:
Can you please point me how it is defined/used in case of iOS?

Reply:
for iOS I thought we could specify it in the Build phase. But now I feel this is a bit too hard to implement.
Now there is an `.xcode.env` file in the `ios` folder with env variables that are going to be used for the build.

I thought we could do a `.datadog.sourcemaps.env` file at the root of the project, which could contain variables like:
- DATADOG_API_KEY
- DATADOG_SITE
- DATADOG_ANDROID_SERVICE
- DATADOG_IOS_SERVICE

This file can be gitignored so the variables could also be specified in the CI without having to create or decrypt a file.

In the plugin, we could try to source it if is present, then use the env variable that is present.
I think there is an advantage of having only 1 file for iOS and Android, and it integrates nicely with the CI, Fastlane or the `.xcode.env` (we can add a line to source our file from this one).

What do you think?

Reply:
the only downside is that not everything should be sourced (service name is not read as env variable by `datadog-ci`, if I'm not wrong?) and there is no `source` on Windows (but it is not a big deal, with Gradle we can source by ourselves), but overall I'm ok with such common file.

Reply:
Indeed the service name is not read as an environment variable by datadog-ci, I don't think it should be (if they want to directly use this command, users can create a script that gets this variable and feeds it into the script).

I think the advantage of having 1 config file is that it makes it easy for devs to know where to put things.